### PR TITLE
feature (webapi): add support for newton ontology

### PR DESCRIPTION
**Pull request by SepidehAlassi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -57,9 +57,9 @@ jobs:
         - rm -rf $HOME/.ivy2
         - cd $TRAVIS_BUILD_DIR && sbt clean stage
         # only uncomment if docker image is needed for this branch. remove before merging into develop
```
Comment (benjamingeer) [inline .travis.yml]: The comments says "remove before merging into develop". :)

File: webapi/_test_data/ontologies/newton-onto.ttl [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# RESOURCES
+#
+##########################################################
+
+### ###########################################
+### newton:letter
+
+:letter rdf:type owl:Class ;
+
+        rdfs:subClassOf beol:letter ,
```
Comment (benjamingeer) [inline webapi/_test_data/ontologies/newton-onto.ttl]: Would it make more sense to have a shared base class of `newton:letter` and `beol:letter`, rather than making `newton:letter` a subclass of `beol:letter`? See [Shared Ontologies](https://docs.knora.org/paradox/02-knora-ontologies/introduction.html#shared-ontologies).

File: .travis.yml (hunk 1 above)
Comment (SepidehAlassi) [reply .travis.yml]: haha, Ivan has ignored this comment 😄 I will fix it

**Comment by tobiasschweizer** [comment]
@SepidehAlassi Could develop be merged into this branch so it is up to date? I would like to make some checks with the beol data.

**Comment by benjamingeer** [comment]
I'll let you and @tobiasschweizer sort out whether `newton:letter` should be a subclass of `beol:letter`.

**Comment by tobiasschweizer** [comment]
@SepidehAlassi I am available on Skype this afternoon to discuss this (
https://github.com/dhlab-basel/Knora/pull/1193#issuecomment-460974262).

**Comment by SepidehAlassi** [comment]
@tobiasschweizer are you using the property letterHasLanguage in BEBB?

**Comment by tobiasschweizer** [comment]
> are you using the property letterHasLanguage in BEBB?

No. But I could actually think about using it since the language is indicated in the metadata.

**Comment by SepidehAlassi** [comment]
@tobiasschweizer I am gonna just change it to "hasLanguage" and make it a property of "writtenSource"

**Comment by SepidehAlassi** [comment]
@tobiasschweizer I left the property to be "letterHasLanguage" but made it a property of the abstract class "basicLetter". You can easily use it for BEBB letters

**Comment by tobiasschweizer** [comment]
> I left the property to be "letterHasLanguage" but made it a property of the abstract class "basicLetter". You can easily use it for BEBB letters

Ok!

@SepidehAlassi I assume that the changes introduced by this PR all work with the current data (last import which is on the test server) and no changes are required in the import routines. With these changes, you minimized the redundancy in the ontology. Did I understand that correctly?

**Comment by SepidehAlassi** [comment]
@benjamingeer does the ontology look better now? 😉

**Comment by SepidehAlassi** [comment]
@tobiasschweizer yes, you understand it correctly. no change is needed in data or import routine

File: webapi/_test_data/ontologies/newton-onto.ttl (hunk 2 above)
Comment (SepidehAlassi) [reply webapi/_test_data/ontologies/newton-onto.ttl]: done :)

File: webapi/_test_data/ontologies/biblio-onto.ttl [hunk 3]
```diff
@@ -92,12 +92,12 @@
 
                     rdfs:comment """Represents the location of a publication."""@en ;
 
+                    knora-base:subjectClassConstraint knora-base:Resource ;
```
Comment (benjamingeer) [inline webapi/_test_data/ontologies/biblio-onto.ttl]: Why did you change this from `:Publication` to `:Resource`?

File: webapi/_test_data/ontologies/beol-onto.ttl [hunk 4]
```diff
@@ -1532,9 +1538,9 @@
 
 
 ### ###########################################
-### beol:letter
+### beol:basicLetter
 
-:letter rdf:type owl:Class ;
+:basicLetter rdf:type owl:Class ;
```
Comment (benjamingeer) [inline webapi/_test_data/ontologies/beol-onto.ttl]: I like `basicLetter`, but I'm wondering if it really belongs in the `beol` ontology. Maybe it would make sense to put it in the `biblio` ontology?

File: webapi/_test_data/ontologies/biblio-onto.ttl (hunk 3 above)
Comment (SepidehAlassi) [reply webapi/_test_data/ontologies/biblio-onto.ttl]: hmm because I had used this property in Newton ontology for another resource. Then I have got rid of that resource, but this subjectClassConstraint has remained like this.  :/

File: webapi/_test_data/ontologies/beol-onto.ttl (hunk 4 above)
Comment (SepidehAlassi) [reply webapi/_test_data/ontologies/beol-onto.ttl]: no, in my opinion, it should stay in beol ontology because in Biblio Ontology we just have the items which appear in a bibliography list. On the other hand, changing this will cause a loop in loading ontologies. On the other hand, we then would need to immensely change the structure of data and letter templates

Comment (benjamingeer) [reply webapi/_test_data/ontologies/beol-onto.ttl]: OK, no problem.

File: webapi/_test_data/ontologies/biblio-onto.ttl (hunk 3 above)
Comment (benjamingeer) [reply webapi/_test_data/ontologies/biblio-onto.ttl]: I think it's clearer to make it more specific if you can.

Comment (SepidehAlassi) [reply webapi/_test_data/ontologies/biblio-onto.ttl]: done!

**Comment by SepidehAlassi** [comment]
@benjamingeer can I merge this now?

**Comment by benjamingeer** [comment]
Yes please. :)
